pi_futex.h

```diff
@@ -0,0 +1,58 @@
+#ifndef PI_FUTEX_H
```

Comment:
This should either precede the previous patch, or be merged I presume? I was wondering if we should use the futextest.h header I wrote for the linux kernel futex tests (tools/testing/selftests/futex/include/futextest.h) but your version is better suited to this library, so we'll use yours.